common/models/entities_with_tags.go

```diff
@@ -0,0 +1,43 @@
+package models
+
+type EntitiesWithTags []*EntityWithTags
+
+type ewtFilter func(e EntityWithTags) bool
+
+func (e EntitiesWithTags) RemoveIf(f ewtFilter) EntitiesWithTags {
```

Comment:
I think I'm with zp on this one - it feels similar to some backend design practices I've picked up in Python. Example:
```
class ExampleModel(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(256))
    created = db.Column(db.DateTime, nullable=False, default=datetime.utcnow())

    @classmethod
    def new(cls, name, session=None):
        if session is None:
            session = db.session
        instance = cls(name=name)
        session.add(instance)
        session.commit()
        return instance

    @classmethod
    def get_by_name(cls, name, session=None):
        if session is None:
            session = db.session
        try:
            instance = session.query(cls).filter_by(string_id=string_id).one()
        except NoResultFound:
            instance = None
        return instance
```

This way, each model know things about how it should go about creating/deleting/querying itself. If you've got multiple modules all accessing the same model, it prevents you from needing to define that logic in multiple places. Of course, you folks already get around that by using the `common/` dir, but it just feels cleaner to me for model logic to be with the models, instead of divorced and defined somewhere completely different.

</twocents>